test/unit/range/iterator_test_template.hpp

```diff
@@ -0,0 +1,518 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <seqan3/std/iterator>
+
+using namespace seqan3;
+
+template <typename T>
+struct iterator_fixture : public ::testing::Test
```

Comment:
So I checked out this PR and tested one of my current iterators. Since it has a more complex setting the entire setup is really not comfortable. I am also required to provide the iterator type to specialise the iterator_fixture in order to make this test case work, because it uses TypeParam in the concept checks. Now the solution that I propose remedies all of this struggle for the test developer. Moreover, in my proposal I can reuse the test fixture for different specialisations of the same iterator. For instance I have an iterator that changes it's behavior given a policy. I can define the iterator inside the fixture with the given policy as type argument to the test instantiation and do not need to provide them globally or outside of the class.  It would be even possible to provide a base class that can be used to customise the fixture behavior for example when additional steps are needed in the fixture methods SetUp or TearDown. Moreover, if I have additional custom tests to the iterator or other components, like if I have a range and strongly coupled an iterator with it, I might as well test both, the iterator and additionally the range while reusing the same test fixture/data.

Reply:
Ok I give up. Since you have a more complicated use case, why don't you supersede this PR and adjust the code to your needs?